compiler/rustc_mir_transform/src/check_const_item_mutation.rs

```diff
@@ -53,9 +53,10 @@ impl<'tcx> ConstMutationChecker<'_, 'tcx> {
         //
         //     #[const_mutation_allowed]
         //     pub const LOG: Log = Log { msg: "" };
-        match self.tcx.adt_destructor(def_id) {
-            Some(_) => None,
-            None => Some(def_id),
+        match self.tcx.type_of(def_id).skip_binder().ty_adt_def().map(|adt| adt.has_dtor(self.tcx))
```

Comment:
previously this was using the `def_id` of the const, which accidentally worked because the query invoked `type_of` on it internally.